sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/StreamingProducer.cs

```diff
@@ -0,0 +1,377 @@
+        /// <param name="eventHubName"></param>
+        /// <param name="streamingOptions"></param>
+        public StreamingProducer(string connectionString, string eventHubName = default, StreamingProducerOptions streamingOptions = default)
+        {
+            Producer = new EventHubProducerClient(connectionString, eventHubName, streamingOptions);
+            _streamingProducer(streamingOptions);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingProducer" /> class.
+        /// </summary>
+        /// <param name="fullyQualifiedNamespace"></param>
+        /// <param name="eventHubName"></param>
+        /// <param name="credential"></param>
+        /// <param name="streamingOptions"></param>
+        public StreamingProducer(string fullyQualifiedNamespace, string eventHubName, AzureNamedKeyCredential credential, StreamingProducerOptions streamingOptions = default)
+        {
+            Producer = new EventHubProducerClient(fullyQualifiedNamespace, eventHubName, credential, streamingOptions);
+            _streamingProducer(streamingOptions);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingProducer" /> class.
+        /// </summary>
+        /// <param name="fullyQualifiedNamespace"></param>
+        /// <param name="eventHubName"></param>
+        /// <param name="credential"></param>
+        /// <param name="streamingOptions"></param>
+        public StreamingProducer(string fullyQualifiedNamespace, string eventHubName, AzureSasCredential credential, StreamingProducerOptions streamingOptions = default)
+        {
+            Producer = new EventHubProducerClient(fullyQualifiedNamespace, eventHubName, credential, streamingOptions);
+            _streamingProducer(streamingOptions);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingProducer" /> class.
+        /// </summary>
+        /// <param name="fullyQualifiedNamespace"></param>
+        /// <param name="eventHubName"></param>
+        /// <param name="credential"></param>
+        /// <param name="streamingOptions"></param>
+        public StreamingProducer(string fullyQualifiedNamespace, string eventHubName, TokenCredential credential, StreamingProducerOptions streamingOptions = default)
+        {
+            Producer = new EventHubProducerClient(fullyQualifiedNamespace, eventHubName, credential, streamingOptions);
+            _streamingProducer(streamingOptions);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StreamingProducer" /> class.
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="streamingOptions"></param>
+        public StreamingProducer(EventHubConnection connection, StreamingProducerOptions streamingOptions = default)
```

Comment:
```suggestion
        public StreamingProducer(EventHubConnection connection, StreamingProducerOptions clientOptions = default)
```

That name is required by the guidelines.